source/Nevermore/Util/DataModificationQueryBuilder.cs

```diff
@@ -457,7 +491,19 @@ class RelatedDocumentTableData
 
     internal static class DataModificationQueryBuilderExtensions
     {
+        static readonly Dictionary<Type, string> IdentityIdTypeMap = new Dictionary<Type, string>
+        {
+            [typeof(short)] = "smallint",
+            [typeof(int)] = "int",
+            [typeof(long)] = "bigint"
```

Comment:
because we have to define a `TABLE` row version, we need the sql type that matches our `Id` column.